### PR TITLE
Update consensus so that verify_waitcertificate can get relevant poet pub key from validator-registry-tp

### PR DESCRIPTION
**Pull request by askmish**

### CODE REVIEW

File: src/validator-registry-tp/src/validator_registry_tp.rs [hunk 1]
```diff
@@ -204,10 +204,10 @@ impl ValidatorRegistryTransactionHandler {
                     originator_public_key_hash: &String,
                     val_reg_payload: &ValidatorRegistryPayload,
                     context: &mut TransactionContext,) -> Result <(), ValueError> {
-         let signup_info : ValidatorRegistrySignupInfo = serde_json::from_str(val_reg_payload.signup_info_str).unwrap();
+         let signup_info : ValidatorRegistrySignupInfo = serde_json::from_str(&val_reg_payload.signup_info_str).unwrap();
          let proof_data : SignupInfoProofData = serde_json::from_str(&*signup_info.proof_data).unwrap();
 
-         let report_public_key_pem = self._get_config_setting(context, "sawtooth.poet.report_public_key_pem").unwrap();
+         let report_public_key_pem = self._get_config_setting(context, &"sawtooth.poet.report_public_key_pem".to_string()).unwrap();
```
Comment (arsulegai) [inline src/validator-registry-tp/src/validator_registry_tp.rs]: I see in PoET2 core, these settings have "sawtooth.consensus.poet2." prefix but we are using "sawtooth.poet." prefix (same as what followed in PoET 1). It's better to keep them consistent if possible.

File: docker-compose-3-nodes.yaml [hunk 2]
```diff
@@ -16,11 +16,75 @@
 version: '3.6'
 
 volumes:
-  cargo-registry:
+  cargo-registry1:
+  cargo-registry2:
+  cargo-registry3:
+  cargo-registry11:
```
[the remark this answers is not in the document]
Comment (askmish) [reply docker-compose-3-nodes.yaml]: We could have, but that's not the scope of this build file.

File: docker-compose-3-nodes.yaml [hunk 3]
```diff
@@ -16,11 +16,75 @@
...
+      dockerfile: ./validator-registry-tp.dockerfile
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+    volumes:
+      - ./src/validator-registry-tp:/project/validator-registry-tp/
+      - cargo-registry22:/root/.cargo/registry
+    working_dir: '/project/validator-registry-tp'
+    entrypoint: "bash -c \"rm -rf /project/validator-registry-tp/bin &&\
```
[the remark this answers is not in the document]
Comment (askmish) [reply docker-compose-3-nodes.yaml]: This code has been updated in an earlier PR.

File: docker-compose.yaml [hunk 4]
```diff
@@ -81,6 +103,9 @@ services:
       - validator-1
     entrypoint: "bash -c \"\
         source /project/sawtooth-poet2/build_sgx_binary.sh \
+        && cp /project/sawtooth-poet2/src/validator-registry-tp/src/validator_registry_validator_info.rs /project/sawtooth-poet2/src/core/src \
```
[the remark this answers is not in the document]
Comment (askmish) [reply docker-compose.yaml]: This just simplifies the code and work needed to get the code up.

File: src/validator-registry-tp/src/validator_registry_tp.rs [hunk 5]
```diff
@@ -229,10 +229,10 @@ impl ValidatorRegistryTransactionHandler {
         let entries_ = context.get_state(vec![address.to_string()]); // this return Vec<u8>
         let entries = if entries_.is_ok() {
                                    entries_.unwrap()
-                               } else {
-                                  warn!("Could not get context for address : {}", address);
-                                  panic!("Error getting context.");
-                               };
+                      } else {
+                           warn!("Could not get context for address : {}", address);
+                           panic!("Error getting context.");
```
[the remark this answers is not in the document]
Comment (askmish) [reply src/validator-registry-tp/src/validator_registry_tp.rs]: get state should never fail.

File: src/validator-registry-tp/src/validator_registry_tp.rs [hunk 6]
```diff
@@ -270,9 +270,9 @@ impl ValidatorRegistryTransactionHandler {
         let setting_data = self._get_setting_data(context, &config_key_address);
 
         match setting_data {
-	    None => Ok(None),
-	    Some(entries) => {
-                let setting: Setting = unpack_data(&entries)?;
+	    Err(err) => Err(err),
+	    Ok(entries) => {
+                let setting: Setting = unpack_data(&entries.unwrap())?;
```
[the remark this answers is not in the document]
Comment (askmish) [reply src/validator-registry-tp/src/validator_registry_tp.rs]: Done in another PR raised earlier

File: src/validator-registry-tp/src/validator_registry_tp.rs [hunk 7]
```diff
@@ -322,7 +322,7 @@ fn _config_key_to_address(key : &String) -> String {
     let mut addr_parts : Vec<String> = key_parts.iter().map(|key_part| _config_short_hash(key_part.to_string())).collect();
     let addr_parts_len = addr_parts.len();
     for i in 0..(_CONFIG_MAX_KEY_PARTS - addr_parts_len) {
-        addr_parts.push(_CONFIG_ADDRESS_PADDING);
+        addr_parts.push(_CONFIG_ADDRESS_PADDING.clone());
```
[the remark this answers is not in the document]
Comment (askmish) [reply src/validator-registry-tp/src/validator_registry_tp.rs]: The copying of original repo didn't go well in that case.

File: src/core/src/service.rs [hunk 8]
```diff
@@ -243,6 +243,15 @@ impl Poet2Service {
         self.get_setting( head_id, key )
     }
 
+    pub fn get_state(&mut self, block_id: BlockId, key: &String) -> Result<HashMap<String, Vec<u8>>, Error> {
+        self.service.get_state(
+            block_id,
+            vec![
+                     key.clone(),
+                ]);
```
[the remark this answers is not in the document]
Comment (askmish) [reply src/core/src/service.rs]: Please get yourself familiar with the sdk code.

**Comment by askmish** [comment]
@arsulegai Please be aware of the context in which you're reviewing this PR. An earlier PR(that you've raised) is still in review and is addressing many of those comments you've repeated here. I have also updated the PR title to be more clear on its purpose.

File: docker-compose.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (manojgop) [reply docker-compose.yaml]: @askmish Can we use validator reg tp crate in core and then use this struct in core. This seems better option than copying files

File: src/core/src/service.rs (hunk 8 above)
[the remark this answers is not in the document]
Comment (arsulegai) [reply src/core/src/service.rs]: Ok, even this as comment works.
But please add a comment saying why we call service.get_state() but then return Ok(empty-hashmap)? Why not return result of service.get_state()?

File: src/validator-registry-tp/src/validator_registry_tp.rs (hunk 7 above)
[the remark this answers is not in the document]
Comment (arsulegai) [reply src/validator-registry-tp/src/validator_registry_tp.rs]: I see same code in original repo

File: src/validator-registry-tp/src/validator_registry_tp.rs (hunk 1 above)
Comment (arsulegai) [reply src/validator-registry-tp/src/validator_registry_tp.rs]: I see same code in original poet2 repo

Comment (askmish) [reply src/validator-registry-tp/src/validator_registry_tp.rs]: It seems the repo switch happened before those changes could be pushed.

Comment (arsulegai) [reply src/validator-registry-tp/src/validator_registry_tp.rs]: Ok, no issues. Please push those changes to new repo. So copy happened correct but code was not pushed.

File: src/validator-registry-tp/Cargo.toml [hunk 9]
```diff
@@ -4,7 +4,7 @@ version = "1.0.0"
 authors = ["Intel Corporation"]
 
 [dependencies]
-sawtooth_sdk = { git = "https://github.com/hyperledger/sawtooth-core.git" }
+sawtooth_sdk = { git = "https://github.com/hyperledger/sawtooth-core.git", rev="7ac644d" }
```
[the remark this answers is not in the document]
Comment (askmish) [reply src/validator-registry-tp/Cargo.toml]: The code was relevant at the time when the PR was raised. The changes have just gone in yesterday, the new changes will be picked up during rebase.

File: src/core/src/service.rs [hunk 10]
```diff
@@ -243,6 +243,15 @@ impl Poet2Service {
...
+       Ok(HashMap::new())
```
Comment (askmish) [inline src/core/src/service.rs]: This code is a stub, to be removed during final integration.

**Comment by arsulegai** [comment]
I've cherry-picked changes related to reading PoET public key in PR https://github.com/manojgop/sawtooth-poet/pull/7

**Comment by arsulegai** [comment]
These commits are cherry-picked

**Pull request closed without merging**